src/JustEat.StatsD/SocketTransport.cs

```diff
@@ -29,6 +29,11 @@ public sealed class SocketTransport : IStatsDTransport, IStatsDBufferedTransport
         public SocketTransport(IPEndPointSource endPointSource, SocketProtocol socketProtocol)
         {
             _endpointSource = endPointSource ?? throw new ArgumentNullException(nameof(endPointSource));
+
+            if (socketProtocol != SocketProtocol.IP && socketProtocol != SocketProtocol.Udp)
+            {
+                throw new ArgumentOutOfRangeException(nameof(socketProtocol));
```

Comment:
If you use [this constructor](https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception.-ctor?view=netframework-4.7.2#System_ArgumentOutOfRangeException__ctor_System_String_System_Object_System_String_) the offending value can be included to improve the debugging experience.